File: MovieCatalogQw/app/src/main/java/com/unlistedi/moviecatalogqw/ui/tvshows/TvShowsAdapter.java
package com.unlistedi.moviecatalogqw.ui.tvshows;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.unlistedi.moviecatalogqw.DetailActivity;
import com.unlistedi.moviecatalogqw.R;
import com.unlistedi.moviecatalogqw.data.MovieEntity;

import java.util.ArrayList;

public class TvShowsAdapter extends RecyclerView.Adapter<TvShowsAdapter.TvShowsViewHolder> {
    private ArrayList<MovieEntity> tvShows;

    public TvShowsAdapter(ArrayList<MovieEntity> tvShows) {
        this.tvShows = tvShows;
    }


    @NonNull
    @Override
    public TvShowsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new TvShowsViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_movie, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull final TvShowsViewHolder holder, final int position) {
        holder.tvListTitle.setText(tvShows.get(position).getTitle());
        holder.tvListScore.setText(String.valueOf(tvShows.get(position).getScore()));
        Glide.with(holder.itemView.getContext())
                .load(tvShows.get(position).getImg_path())
                .into(holder.ivPoster);
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(holder.itemView.getContext(), DetailActivity.class);
                intent.putExtra(DetailActivity.EXTRA_MOVIE, tvShows.get(position));
                holder.itemView.getContext().startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
        return tvShows.size();
    }

    class TvShowsViewHolder extends RecyclerView.ViewHolder{
        TextView tvListTitle, tvListScore;
        ImageView ivPoster;

        TvShowsViewHolder(@NonNull View itemView) {
            super(itemView);
            tvListTitle = itemView.findViewById(R.id.tvListTitle);
            tvListScore = itemView.findViewById(R.id.tvListScore);
            ivPoster = itemView.findViewById(R.id.ivPoster);
        }
    }
}

File: MovieCatalogueKu/app/src/main/java/com/unlistedi/moviecatalogueku/ui/tvshows/TvShowsAdapter.kt
package com.unlistedi.moviecatalogueku.ui.tvshows

import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.unlistedi.moviecatalogueku.DetailActivity
import com.unlistedi.moviecatalogueku.R
import com.unlistedi.moviecatalogueku.data.MovieEntity
import com.unlistedi.moviecatalogueku.utils.GlideApp

class TvShowsAdapter(private val tvShows : ArrayList<MovieEntity>) : RecyclerView.Adapter<TvShowsAdapter.TvShowsViewHolder>() {
    class TvShowsViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvListTitle : TextView = itemView.findViewById(R.id.tvListTitle)
        val tvListScore : TextView = itemView.findViewById(R.id.tvListScore)
        val ivPoster : ImageView = itemView.findViewById(R.id.ivPoster)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TvShowsViewHolder {
        return TvShowsViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.items_movie, parent,false))
    }

    override fun getItemCount(): Int {
        return  tvShows.size
    }

    override fun onBindViewHolder(holder: TvShowsViewHolder, position: Int) {
        holder.tvListTitle.text = tvShows[position].title
        holder.tvListScore.text = tvShows[position].score.toString()

        GlideApp.with(holder.itemView.context)
            .load(tvShows[position].img_path)
            .into(holder.ivPoster)
        holder.itemView.setOnClickListener {
            val intent = Intent(holder.itemView.context, DetailActivity::class.java)
            intent.putExtra(DetailActivity.EXTRA_MOVIE, tvShows[position])
            holder.itemView.context.startActivity(intent)
        }
    }
}